### ISSUE TITLE
Add og:url metatag to help Facebook identify a canonical URL for each page

### ISSUE THREAD

**Issue by jalperin** [pkp/pkp-lib#3250]
The og:url tag should appear on all article pages, and should probably point to the abstract page for each. This way FB resolves all instances of an article the same way. Alternative, it could point to the same URL as the DOI (if that's different).  There are other og: tags that could accompany the articles. 

[Some additional information on Facebook metatags](https://developers.facebook.com/docs/sharing/webmasters#markup)

**Comment by Vitaliy-1** [comment]
I think it would be ideal to make something like a social meta plugin for facebook and twitter meta-tags. it is quite easy to implement.

**Comment by Vitaliy-1** [comment]
I could make "social-meta plugin" if there is a need. It is really a matter of 1-2 hours.

**Comment by jalperin** [comment]
I think so, except the metatags should also be present on the galley pages, not just on the article landing page.

**Comment by Vitaliy-1** [comment]
@NateWr @jalperin 
Made a plugin (only for article page for now) that adds Open Graph and Twitter meta-tags. Any suggestions on this?
https://github.com/Vitaliy-1/socialMeta

**Comment by NateWr** [comment]
:+1: cool @Vitaliy-1! A couple of suggestions for improvements:

- Add for issues, announcements, and homepage.
- If no article cover image exists, fall back to issue cover image.
- Add og:locale if the journal supports multiple locales.
- For the header IDs, use the meta IDs. So `ogURL` should be `og:url`. This will ensure there's never any duplicates or conflicts if people add other header data.

And a really big one that would be great would be for the tags to be hooked themselves so developers can tailor them as they'd liike. I'd recommend building them as an assoc array and then converting to headers like this:

```
$socialHeaders = array(
	'og:url' => 'meta property="og:url" content="' . $request->url(null, 'article', 'view', array($article->getBestArticleId())) . '"/>',
	...
);
HookRegistry::call('SocialHeaders::Article', array(&$socialHeaders, $article, $request));
foreach ($socialHeaders as $key => $socialHeader) {
	$templateMgr->addHeader($key, $socialHeader);
}
```

**Comment by Vitaliy-1** [comment]
Sorry for delay. Will return for this issue only after finishing JATS WYSIWYG online editor.

**Comment by NateWr** [comment]
Closing this as outdated. If you feel this is still important, please consider making a proposal in the [feature request category](https://forum.pkp.sfu.ca/c/feature-requests/8) of our community forum.

**Comment by jalperin** [comment]
Shouldn't this be turned into a new entry for the suggested plugin? Seems
like plugin is good idea, no?

**Comment by asmecher** [comment]
@jalperin, this is an unscheduled and unprioritized feature request, and we're tracking those on the [feature request category](https://forum.pkp.sfu.ca/c/feature-requests/8) of the forum rather than here. If this is still a priority for you, I'd suggest making a post there, and adding a link to it from the Scientific/Research project.

**Comment by jalperin** [comment]
Vitaliy implemented this in a plugin back in 2018, but it was never included in the Plugin Gallery. Before closing, couldn't that be reviewed and considered?

**Comment by asmecher** [comment]
No, that would mean diverting someone from what they're working on to treat this as a higher priority.

**Comment by jalperin** [comment]
I don't understand why we're closing this entry. I understand it is not a priority, but there is a documented feature, there is an initial implementation, and feedback from Nate. Couldn't this be flagged with a "try me" tag?

To be clear, it is not a huge priority for me, but it is important. Most people wouldn't have any idea is possible because the need for it is not clear to someone who isn't technical, so it will never get a lot of votes, but it is also super simple to implement.

**Comment by asmecher** [comment]
The process is outlined here: https://pkp.sfu.ca/2022/07/20/improving-how-we-collect-community-feedback/
Votes aren't the only consideration on the forum, especially if it also gets tracked in the Scientific/Research project.

**Comment by jalperin** [comment]
[Posted in Forum](https://forum.pkp.sfu.ca/t/add-og-url-metatag-to-help-facebook-identify-a-canonical-url-for-each-page/75015)

**Comment by NateWr** [comment]
The [openGraph](https://github.com/ajnyga/openGraph) plugin provides this feature. It's built by @ajnyga and can be found in the plugin gallery for 3.2.x and 3.3.x (at the time of writing).
